docs/adr/0045-testing-updates-and-recommendations.md

```diff
@@ -0,0 +1,233 @@
+
+The dashboard has an opportunity to push metrics about test health into the limelight.
+
+</details>
+
+<details>
+<summary>D: Update and Expand Central Test Documentation</summary>
+
+Options:
+
+1. Revise and Extend Existing Docs **👍 Recommended**
+2. Minimal Edits for Accuracy
+
+#### Details
+
+We have most documentation around testing in the [developer docs], however test references and patterns are spread about throughout the docs, and we're missing guidance on how to write good tests. As a result, most patterns of testing come down to personal preference, or who is doing a code review. We try to stick to patterns but it can always vary.
+
+Additionally, documentation around _what_ testing exists is spread around and may be difficult to find. It's okay to have them spread and closer to the thing they should be, but also linking test documentation back to a single source could make them easier to find and identify areas that we're missing.
+
+At a minimum however, the documentation we do have may be out of date, or include reference we have not fully adopted, for example the [Test Strategy][test strategy] document. We should review all test related documents, update references and links, and if there are things yet to be fully adopted we should
+
+#### Pros and Cons of each option
+
+- Revise and Extend Existing Docs
+  - **Good**, because we get a single "home" for testing in FxA
+  - **Good**, because participating in testing, be that unit, integration, or functional, becomes easier when there is guidance on how to add and update tests
+  - **Good**, because we also get a place for best practices and patterns, such as how to identify code-smell when making tests or avoiding too many assertions per test
+  - **Bad**, because it could be come tedious to manage references and keeping testing documentation in sync with latest patterns
+  - **Bad**, because this might be too "strict" and put too many guard rails on testing implementation.
+- Minimal Edits for Accuracy
+  - **Good**, because it's less invasive and focuses on bring any test documentation up to date
+  - **Good**, because it would take _far_ less time and gives us a jumping point to improve patterns and documentation long term
+  - **Bad**, because we might never combe back to it
+  </details>
+
+## Decision Outcome
+
+The following improvements are recommended to enhance the consistency, reliability, and maintainability of testing across the monorepo:
+
+A. Unify Testing Frameworks
+
+      Decision: Proceed with a pilot migration of one or two high-traffic sub-projects from Mocha to Jest. Alternatively, we could tackle low traffic projects or non-mocha test suites. Either way, we start
+
+      As of 4/16/25, checking the last `500` commits, these are the highest traffic (commit counts) projects in `packages`
+      Commits  Subfolder
+      -------  ---------
+           69  fxa-auth-server      <- Candidate one
+           66  fxa-settings         <- Already using Jest
```

Comment:
I can't recall if it was fxa-settings or fxa-payments-server (or both), but do you have an opinion on Jest within the confines of create-react-app?  Specifically, I wanted to change testing env to JSDOM but was unable to do so because of the lock-in with CRA.  I would have imagined Jest tests would operate independently.  Breaking it out of CRA is probably out of scope, but might be something to keep in mind since to me, "already using Jest" and "already using Jest without CRA lock-in" are distinct.

Reply:
@chenba ejected create-react-app from settings in https://github.com/mozilla/fxa/pull/15786
Tickets for that task are still TODO for admin-panel (`FXA-8395`) and payments (`FXA-8394`)

Reply:
Look we're all trying to find the guy who did it

Reply:
Hey, it's a positive thing. Credit where credit is due